File: src/dependent/fulfillment/order_fulfillment_service.py
from .inventory_service import InventoryService
from .shipping_service import ShippingService
from .fulfillment_result import FulfillmentResult
from .order import Order


class OrderFulfillmentService:
    def __init__(self, inventory_service: InventoryService, shipping_service: ShippingService) -> None:
        self._inventory_service = inventory_service
        self._shipping_service = shipping_service

    def fulfill(self, order: Order) -> FulfillmentResult:
        reservation_id = self._inventory_service.reserve(order.product_id, order.quantity)
        tracking_id = self._shipping_service.create_shipment(order.order_id, order.address)
        return FulfillmentResult(reservation_id, tracking_id)

File: src/dependent/fulfillment/OrderFulfillmentService.java
package org.example.dependent.fulfillment;

public class OrderFulfillmentService {
    private final InventoryService inventoryService;
    private final ShippingService shippingService;

    public OrderFulfillmentService(InventoryService inventoryService, ShippingService shippingService) {
        this.inventoryService = inventoryService;
        this.shippingService = shippingService;
    }

    public FulfillmentResult fulfill(Order order) {
        String reservationId = inventoryService.reserve(order.getProductId(), order.getQuantity());
        String trackingId = shippingService.createShipment(order.getOrderId(), order.getAddress());
        return new FulfillmentResult(reservationId, trackingId);
    }
}


